REGRESSION (302748@main): Buildbot queues failing to compile WebKit.

bls1999 opened the pull request:
#### 513e0c0df2670f83c2b99a1765f22460f5a5e2b4
<pre>
REGRESSION (302748@main): Buildbot queues failing to compile WebKit.
<a href="https://bugs.webkit.org/show_bug.cgi?id=302184">https://bugs.webkit.org/show_bug.cgi?id=302184</a>
<a href="https://rdar.apple.com/164284543">rdar://164284543</a>

Reviewed by Brianna Fan.

Since the `architectures` property is passed directly into the `compile-webkit`
step in buildbot, changing the join character to a dash instead of a space
broke invocation of `build-webkit`.

This PR changes that behavior such that we only use the dash when uploading.

* Tools/CISupport/build-webkit-org/factories.py:
    (__init__): Change back to using space to join architectures.
* Tools/CISupport/build-webkit-org/steps.py:
    (ConfigureBuild.run): Add new property `archForUpload` that has the correct pattern for uploads.
    (CompileWebKit):
        -&gt; (follow_up_steps): Change to using the new `archForUpload` property to craft upload link.
        -&gt; (evaluateCommand): Ditto.
    (TestMiniBrowserBundle.run): Ditto.
    (RunJavaScriptCoreTests.run): Ditto.
    (RunTest262Tests.run): Ditto.
    (RunWebKitTests.evaluateCommand): Ditto.
    (RunAPITests.run): Ditto.
    (RunWebDriverTests.run): Ditto.
    (ScanBuild.run): Ditto.
* Tools/CISupport/ews-build/steps.py:
    (__init__): Change back to using space to join architectures.
    (ConfigureBuild.run): Add new property `archForUpload` that has the correct pattern for uploads.
    (CompileWebKit):
        -&gt; (follow_up_steps): Change to using the new `archForUpload` property to craft upload link.
        -&gt; (evaluateCommand): Ditto.
    (RunJavaScriptCoreTests.evaluateCommand): Ditto.
    (RunWebKitTests.evaluateCommand): Ditto.
    (RunWebKitTestsInStressMode.evaluateCommand): Ditto.
    (ReRunWebKitTests.evaluateCommand): Ditto.
    (RunWebKitTestsWithoutChange.evaluateCommand): Ditto.
    (RunWebKitTestsRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsRepeatFailuresRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsRepeatFailuresWithoutChangeRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsWithoutChangeRedTree.evaluateCommand): Ditto.
    (RunAPITests.run): Ditto.
    (ScanBuild.uploadLogsSteps): Ditto.
* Websites/webkit.org/wp-content/themes/webkit/build-archives.php: Drive-by fix for incorrect static variable reference.

Canonical link: <a href="https://commits.webkit.org/302754@main">https://commits.webkit.org/302754@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7085c8290dbbe99094248204485d39686e978012

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130076 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/159/builds/2338 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/137473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/81597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131947 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/2305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/156/builds/2229 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/137473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/81597 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133023 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/2305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/137473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/2305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/80745 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/155/builds/2305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/139951 "Build was cancelled. Recent messages:Printed configuration") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/2131 "Build was cancelled. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1; Running apply-patch; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/156/builds/2229 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/139951 "Build was cancelled. Recent messages:Printed configuration") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/129506 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/2176 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/139951 "Build was cancelled. Recent messages:Printed configuration") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/41020 "Build was cancelled. Recent messages:OS: Sonoma (14.7.3), Xcode: 15.4; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit (cancelled)") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/54968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20295 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/158/builds/2204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/2021 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/2221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/2127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -685,6 +685,7 @@ def run(self):
             self.setProperty('configuration', self.configuration, 'config.json')
         if self.architecture:
             self.setProperty('architecture', self.architecture, 'config.json')
+            self.setProperty('archForUpload', '-'.join(self.architecture.split('-')), 'config.json')
```
Do you mean to split `self.architecture` by space here?

bls1999 replied:
Yep, great catch

bls1999 committed:
REGRESSION (302748@main): Buildbot queues failing to compile WebKit.
https://bugs.webkit.org/show_bug.cgi?id=302184
rdar://164284543

Reviewed by Brianna Fan.

Since the `architectures` property is passed directly into the `compile-webkit`
step in buildbot, changing the join character to a dash instead of a space
broke invocation of `build-webkit`.

This PR changes that behavior such that we only use the dash when uploading.

* Tools/CISupport/build-webkit-org/factories.py:
    (__init__): Change back to using space to join architectures.
* Tools/CISupport/build-webkit-org/steps.py:
    (ConfigureBuild.run): Add new property `archForUpload` that has the correct pattern for uploads.
    (CompileWebKit):
        -> (follow_up_steps): Change to using the new `archForUpload` property to craft upload link.
        -> (evaluateCommand): Ditto.
    (TestMiniBrowserBundle.run): Ditto.
    (RunJavaScriptCoreTests.run): Ditto.
    (RunTest262Tests.run): Ditto.
    (RunWebKitTests.evaluateCommand): Ditto.
    (RunAPITests.run): Ditto.
    (RunWebDriverTests.run): Ditto.
    (ScanBuild.run): Ditto.
* Tools/CISupport/ews-build/steps.py:
    (__init__): Change back to using space to join architectures.
    (ConfigureBuild.run): Add new property `archForUpload` that has the correct pattern for uploads.
    (CompileWebKit):
        -> (follow_up_steps): Change to using the new `archForUpload` property to craft upload link.
        -> (evaluateCommand): Ditto.
    (RunJavaScriptCoreTests.evaluateCommand): Ditto.
    (RunWebKitTests.evaluateCommand): Ditto.
    (RunWebKitTestsInStressMode.evaluateCommand): Ditto.
    (ReRunWebKitTests.evaluateCommand): Ditto.
    (RunWebKitTestsWithoutChange.evaluateCommand): Ditto.
    (RunWebKitTestsRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsRepeatFailuresRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsRepeatFailuresWithoutChangeRedTree.evaluateCommand): Ditto.
    (RunWebKitTestsWithoutChangeRedTree.evaluateCommand): Ditto.
    (RunAPITests.run): Ditto.
    (ScanBuild.uploadLogsSteps): Ditto.
* Websites/webkit.org/wp-content/themes/webkit/build-archives.php: Drive-by fix for incorrect static variable reference.

Canonical link: https://commits.webkit.org/302754@main
